src/new_index/mempool.rs

```diff
@@ -300,17 +304,13 @@ impl Mempool {
         }
     }
 
-    fn add(&mut self, txs: Vec<Transaction>) -> Result<()> {
+    fn add(&mut self, txs_map: HashMap<Txid, Transaction>) -> Result<()> {
```

Comment:
is this function  called just from `add_by_txid` ?
If so this map always contains one value,  thus I would accept singular parameter txid, tx and then initiate the map internally.

Reply:
No, it is also used by `Mempool::update()` with multiple transactions.